src/seccomp/src/bin.rs

```diff
@@ -0,0 +1,567 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+mod compiler;
+mod syscall_table;
+
+use bincode::Error as BincodeError;
+use compiler::{Compiler, Error as FilterFormatError, Filter};
+pub use seccomp::BpfThreadMap;
+use serde_json::error::Error as JSONError;
+use std::collections::HashMap;
+use std::fmt;
+use std::fs::File;
+use std::io;
+use std::io::{BufReader, Read};
+use std::path::PathBuf;
+use std::process;
+use utils::arg_parser::{ArgParser, Argument, Arguments as ArgumentsBag};
+
+const SECCOMPILER_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+const SUPPORTED_ARCHES: [&str; 2] = ["x86_64", "aarch64"];
+
+#[derive(Debug)]
+pub enum Error {
+    Bincode(BincodeError),
+    FileOpen(PathBuf, io::Error),
```

Comment:
I feel like the `FileOpen` message is more descriptive. This error can only happen on opening files, and this information can be helpful to the user.
If we rename the error to `IoError`, it would also make sense to modify the message to a general IO error, which would be less descriptive in this case.
WDYT?